applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -71,46 +71,49 @@ void BaseSolidElement::Initialize()
 
 void BaseSolidElement::InitializeSolutionStep( ProcessInfo& rCurrentProcessInfo )
 {
-    const SizeType number_of_nodes = GetGeometry().size();
-    const SizeType dimension = GetGeometry().WorkingSpaceDimension();
-    const SizeType strain_size = mConstitutiveLawVector[0]->GetStrainSize();
+    // We initialize the material reponse if required
+    if (mConstitutiveLawVector[0]->RequiresInitializeMaterialResponse()) {
```

Comment:
why only checking the first CLaw? => this is a bug if you have different CLaws for different GPs

Reply:
OK